### PR TITLE
Protected create

### PR DESCRIPTION
**Pull request by RyanJ2463**

### CODE REVIEW

File: app/javascript/components/App.js [hunk 1]
```diff
@@ -13,6 +13,36 @@ import SongShow from './pages/SongShow'
...
+    this.state = {
+      songs: []
+    }
+  }
+  componentDidMount(){
+    this.readSong()
+  }
+
+  createSong = (newSong) => {
+    fetch("http://localhost:3000/songs", {
```
Comment (sjproctor) [inline app/javascript/components/App.js:27]: Since all the code is coming from the same base url you can fetch to just your endpoint `/songs`

**Review by SunkissedQueen — APPROVED** [review]
Great work on introducing a new concept for the group. Keep that devise link bookmarked

File: app/javascript/components/App.js [hunk 2]
```diff
@@ -13,6 +13,36 @@ import SongShow from './pages/SongShow'
...
+      method: "POST"
+    })
+    .then(response => response.json())
+    .then(payload => this.readSong())
+    .catch(errors => console.log("Song create errors:", errors))
+  }
+
+  readSong = () => {
+    fetch("http://localhost:3000/songs")
+    .then(response => response.json())
```
Comment (sjproctor) [inline app/javascript/components/App.js:41]: Same thing here. And you won't have to change this line when you deploy.

File: app/javascript/components/pages/SongNew.js [hunk 3]
```diff
@@ -1,11 +1,62 @@
...
+  };
+
+  handleSubmit = () => {
+    this.props.createSong(this.state.form);
+    this.setState({ submitted: true });
+  };
   render() {
-    return (
-      <div>SongNew</div>
+    console.log(this.state.form);
```
Comment (sjproctor) [inline app/javascript/components/pages/SongNew.js:27]: Once you know the form is doing what you want it to do the console.logs should be removed.

File: spec/requests/songs_spec.rb [hunk 4]
```diff
@@ -2,6 +2,36 @@
...
+        # creating a song with a signed in user
+        song_params = {
+          song: {
+          title: "i miss the old Kayne",
+          artist: "Kayne"}
+        }
+        post '/songs', params: song_params
+        song_response = JSON.parse(response.body)
+        p song_response
+        expect(song_response["title"]).to eq "i miss the old Kayne" #didnt use the index [0] since it wasnt returning an array
```
Comment (sjproctor) [inline spec/requests/songs_spec.rb:33]: Typically you would remove comments from your code before merging to main.

**Review by sjproctor — CHANGES_REQUESTED** [review]
Couple small things to clean up.

**Pull request merged**